### PR TITLE
[widgets] introduce column picker component

### LINKED ISSUE #185: Implement a `ColumnPicker` component
This component will use internally the `AutocompleteWidget` but will do the necessary plumbering on the store so that column selection / changes is handled automatically.

Currently, all step forms will reimplement this.

For instance,

```html
    <WidgetAutocomplete
        id="columnInput"
         v-model="step.column"
         name="Fill null values in:"
        :options="columnNames"
       @input="setSelectedColumns({ column: step.column })"
         placeholder="Enter a column"
   ></WidgetAutocomplete>
```

and all related code / watchers in step forms could be replaced by

```html
    <ColumnPicker
        v-model="step.column"
        id="oldnameInput"
        name="Fill null values in:"
        placeholder="Enter a column"
    ></ColumnPicker>
```

### PR DESCRIPTION
**Pull request by adimascio**
closes #185

### CODE REVIEW

File: src/components/stepforms/ColumnPicker.vue [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  @Watch('selectedColumns')
+  onSelectedColumnsChanged(val: string[], oldVal: string[]) {
+    if (!_.isEqual(val, oldVal)) {
+      this.column = val[0];
+    }
+  }
+
+  created() {
```
Comment (Dalanir) [inline src/components/stepforms/ColumnPicker.vue]: I would put it before `methods` and `watchers`

File: tests/unit/column-picker.spec.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        data: [],
+      },
+    });
+    const wrapper = shallowMount(ColumnPicker, {
+      store,
+      localVue,
+    });
+    expect(wrapper.vm.$data.column).to.be.null;
+    store.commit('setSelectedColumns', { column: 'columnB' });
+    await Vue.nextTick();
```
Comment (Dalanir) [inline tests/unit/column-picker.spec.ts]:
```suggestion
    await localVue.nextTick();
```

File: tests/unit/delete-column-step-form.spec.ts [hunk 3]
```diff
@@ -122,8 +109,8 @@ describe('Delete Column Step Form', () => {
       store,
       localVue,
     });
-    wrapper.setData({ column: 'columnB' });
-    await wrapper.find(WidgetAutocomplete).trigger('input');
+    wrapper.find(WidgetAutocomplete).vm.$emit('input', 'columnB');
```
Comment (vdestraitt) [inline tests/unit/delete-column-step-form.spec.ts]: I think that this piece of code is inn a a test that checks something already tested in column-picker.spec.ts

File: tests/unit/fillna-step-form.spec.ts [hunk 4]
```diff
@@ -126,8 +113,8 @@ describe('Fillna Step Form', () => {
       store,
       localVue,
     });
-    wrapper.setData({ editedStep: { column: 'columnB', value: '' } });
-    await wrapper.find(WidgetAutocomplete).trigger('input');
+    wrapper.find(WidgetAutocomplete).vm.$emit('input', 'columnB');
```
Comment (vdestraitt) [inline tests/unit/fillna-step-form.spec.ts]: idem

File: tests/unit/rename-step-form.spec.ts [hunk 5]
```diff
@@ -154,8 +141,8 @@ describe('Rename Step Form', () => {
       store,
       localVue,
     });
-    wrapper.setData({ editedStep: { name: 'rename', oldname: 'columnB', new_name: '' } });
-    await wrapper.find(WidgetAutocomplete).trigger('input');
+    wrapper.find(WidgetAutocomplete).vm.$emit('input', 'columnB');
```
Comment (vdestraitt) [inline tests/unit/rename-step-form.spec.ts]: idem

**Pull request merged**